app/src/main/java/com/epfl/drawyourpath/userProfileCreation/UserGoalsInitFragment.kt

```diff
@@ -78,18 +89,27 @@ class UserGoalsInitFragment : Fragment(R.layout.fragment_user_goals_init) {
             }
 
             if (test1 && test2 && test3) {
-                database.setUserGoals(username, distanceGoal.toDouble(), timeGoal.toDouble(), nunberOfPathGoal)
-
-                val previousActivity = activity
-                if (previousActivity != null) {
-                    val fragManagement = previousActivity.supportFragmentManager.beginTransaction()
-                    val dataToEndProfileCreationFrag: Bundle = Bundle()
-                    //data to transmit to the UserGoalsInitFragment(username)
-                    dataToEndProfileCreationFrag.putString("userName", username)
-                    val endProfileCreationFrag = EndProfileCreationFragment()
-                    endProfileCreationFrag.arguments = dataToEndProfileCreationFrag
-                    fragManagement.replace(R.id.userGoalInitFragment, endProfileCreationFrag)
-                        .commit()
+                var userLog = FirebaseAuth.getUser()
+                if(isTest){
+                    userLog = MockAuth.MOCK_USER
+                }
+                if(userLog == null) {
+                    throw java.lang.Error("Any user is logged to the app.")
+                }
+                val user = UserModel(userLog, firstname, surname, LocalDate.ofEpochDay(dateOfBirth), distanceGoal.toDouble(),
```

Comment:
There is a problem with how `UserModel` works here. The constructor isn't trivial as it fetches the database for the username. I think that we should modify the constructor to take the username so it doesn't fetch the database whenever it is created.